polaris-react/src/components/TextField/README.md

```diff
@@ -815,6 +815,49 @@ function TextFieldWithSelectTextOnFocusExample() {
 }
 ```
 
+### Text field with the suggestion
+
+<!-- example-for: web -->
+
+Use to provide an autocomplete suggestion inline with the input value. See the combobox component's tag multi-select example for full implementation of the inline autocomplete pattern.
+
+```jsx
+function TextFieldWithSuggestionExample() {
+  const suggestions = ['Ethel', 'Lucy', 'Ricky', 'Fred'];
+
+  const [value, setValue] = useState('');
+  const [suggestion, setSuggestion] = useState('');
+
+  const handleSuggestion = useCallback((nextValue) => {
+    const nextSuggestion = suggestions.find((suggestion) =>
+      suggestion.toLowerCase().startsWith(nextValue.toLowerCase()),
+    );
+
+    if (nextSuggestion) setSuggestion(nextSuggestion);
+  }, []);
+
+  useEffect(() => {
+    if (value !== suggestion) handleSuggestion(value);
+  }, [value]);
+
+  const handleChange = useCallback((value) => {
+    setValue(value);
+    setSuggestion('');
+  }, []);
+
+  return (
+    <TextField
+      focused
+      type="text"
+      label="Customer"
+      value={value}
+      onChange={handleChange}
+      suggestion={suggestion}
```

Comment:
```suggestion
      suggestion={suggestion}
      helpText="Available customer name options are: Ethel, Lucy, Ricky, and Fred"
```

I didn't know what names were possible so including some in help text is a bit more clear.